File: app/src/main/java/fr/erictruong/material/training/ui/components/tabs/ScrollableTabBarActivity.java
package fr.erictruong.material.training.ui.components.tabs;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;

import butterknife.Bind;
import butterknife.ButterKnife;
import fr.erictruong.material.training.R;
import fr.erictruong.material.training.ui.core.adapter.SimpleFragmentPagerAdapter;
import fr.erictruong.material.training.ui.core.fragment.DummyFragment;

public class ScrollableTabBarActivity extends AppCompatActivity {

    @Bind(R.id.scrollable_tab_bar_toolbar) Toolbar toolbar;
    @Bind(R.id.scrollable_tab_bar_tabs) TabLayout tabs;
    @Bind(R.id.scrollable_tab_bar_view_pager) ViewPager viewPager;

    public static void start(Context context) {
        Intent starter = new Intent(context, ScrollableTabBarActivity.class);
        context.startActivity(starter);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_scrollable_tab_bar);
        ButterKnife.bind(this);

        setSupportActionBar(toolbar);
        toolbar.setNavigationIcon(R.drawable.ic_menu_white_24dp);
        toolbar.setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Snackbar.make(toolbar, "Navigation icon", Snackbar.LENGTH_SHORT).show();
            }
        });

        SimpleFragmentPagerAdapter adapter = new SimpleFragmentPagerAdapter(getSupportFragmentManager());
        adapter.addFragment("Item one", DummyFragment.newInstance());
        adapter.addFragment("Item two", DummyFragment.newInstance());
        adapter.addFragment("Item three", DummyFragment.newInstance());
        adapter.addFragment("Item four", DummyFragment.newInstance());
        adapter.addFragment("Item five", DummyFragment.newInstance());
        adapter.addFragment("Item six", DummyFragment.newInstance());
        adapter.addFragment("Item seven", DummyFragment.newInstance());
        adapter.addFragment("Item eight", DummyFragment.newInstance());

        viewPager.setAdapter(adapter);
        tabs.setupWithViewPager(viewPager);
    }
}

File: app/src/main/java/fr/erictruong/material/training/ui/components/tabs/ScrollableTabBarActivity.kt
package fr.erictruong.material.training.ui.components.tabs

import android.os.Bundle
import android.support.v4.view.GravityCompat
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import fr.erictruong.android.core.adapter.SimpleFragmentPagerAdapter
import fr.erictruong.material.training.R
import fr.erictruong.material.training.ui.core.fragment.DummyFragment
import kotlinx.android.synthetic.main.activity_scrollable_tab_bar.scrollable_tab_bar_drawer_layout as drawerLayout
import kotlinx.android.synthetic.main.activity_scrollable_tab_bar.scrollable_tab_bar_tabs as tabs
import kotlinx.android.synthetic.main.activity_scrollable_tab_bar.scrollable_tab_bar_toolbar as toolbar
import kotlinx.android.synthetic.main.activity_scrollable_tab_bar.scrollable_tab_bar_view_pager as viewPager

class ScrollableTabBarActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_scrollable_tab_bar)

        setSupportActionBar(toolbar)
        val toggle = ActionBarDrawerToggle(this, drawerLayout, toolbar, R.string.access_navigation_drawer_open, R.string.access_navigation_drawer_close)
        drawerLayout.addDrawerListener(toggle)
        toggle.syncState()

        val adapter = SimpleFragmentPagerAdapter(supportFragmentManager)
        adapter.addFragment("Item one", DummyFragment())
        adapter.addFragment("Item two", DummyFragment())
        adapter.addFragment("Item three", DummyFragment())
        adapter.addFragment("Item four", DummyFragment())
        adapter.addFragment("Item five", DummyFragment())
        adapter.addFragment("Item six", DummyFragment())
        adapter.addFragment("Item seven", DummyFragment())
        adapter.addFragment("Item eight", DummyFragment())

        viewPager.adapter = adapter
        tabs.setupWithViewPager(viewPager)
    }

    override fun onBackPressed() {
        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
            drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.search_light, menu)
        inflater.inflate(R.menu.settings, menu)
        return true
    }
}
